RecoMuon/TrackerSeedGenerator/interface/SeedMvaEstimatorPhase2.h

```diff
@@ -0,0 +1,88 @@
+#ifndef RecoMuon_TrackerSeedGenerator_SeedMvaEstimatorPhase2_h
+#define RecoMuon_TrackerSeedGenerator_SeedMvaEstimatorPhase2_h
+
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/Utilities/interface/ESInputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeed.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/TrajectorySeed/interface/PropagationDirection.h"
+#include "DataFormats/TrajectoryState/interface/PTrajectoryStateOnDet.h"
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidate.h"
+#include "DataFormats/RecoCandidate/interface/RecoChargedCandidateFwd.h"
+#include "DataFormats/L1TCorrelator/interface/TkMuon.h"
+#include "DataFormats/L1TCorrelator/interface/TkMuonFwd.h"
+#include "DataFormats/L1TMuonPhase2/interface/TrackerMuon.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/TrajectoryState/interface/FreeTrajectoryState.h"
+#include "RecoTracker/TkDetLayers/interface/GeometricSearchTracker.h"
+#include "RecoTracker/TkDetLayers/interface/GeometricSearchTrackerBuilder.h"
+#include "Geometry/TrackerNumberingBuilder/interface/GeometricDet.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include <memory>
+#include <string>
+
+typedef pair<const DetLayer*, TrajectoryStateOnSurface> LayerTSOS;
+typedef pair<const DetLayer*, const TrackingRecHit*> LayerHit;
+
+class GBRForest;
+
+namespace edm {
+  class FileInPath;
+}
+
+class SeedMvaEstimatorPhase2 {
+public:
+  SeedMvaEstimatorPhase2(const edm::FileInPath& weightsfile,
+                         const std::vector<double>& scale_mean,
+                         const std::vector<double>& scale_std);
+  ~SeedMvaEstimatorPhase2();
```

Comment:
Hi, thanks for the suggestion!

However when I make change together with the suggestion below, I get the error while compiling

```
error: 'SeedMvaEstimatorPhase2::~SeedMvaEstimatorPhase2()' marked 'override', but does not override
   47 |   ~SeedMvaEstimatorPhase2() override = default;
```

This is probably because the "SeedMvaEstimatorPhase2" class does not inherit any class and thus it cannot override its mother's virtual destructor? (I'm not sure...)

The original code works fine w/o compile error, so I guess it would be better to stick to the original version.

Reply:
> class does not inherit any class and thus it cannot override its mother's virtual destructor? 

yes, that's the reason. I had thought out of habit it was an EDProducer, but as you point out it's a standalone class.